api/api.go

```diff
@@ -458,9 +460,99 @@ func setExtensionUrls(original core.Extension, rootUrl string) core.Extension {
 		}
 	}
 
+	localization, err := GetLocalization(&extension)
+	if err != nil {
+		panic(err)
+	}
+
+	extension.Localization = localization
+
 	return extension
 }
 
+func GetLocalization(extension *core.Extension) (*core.Localization, error) {
+	path := filepath.Join(".", extension.Development.RootDir, "locales")
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		// The extension does not have a locales directory.
+		return nil, nil
+	}
+
+	fileNames, fileNamesErr := GetFileNames(path)
+	if fileNamesErr != nil {
+		return nil, fileNamesErr
+	}
+	translations := make(map[string]interface{})
+	defaultLocale := ""
+
+	for _, fileName := range fileNames {
+		data, err := GetMapFromFile(filepath.Join(path, fileName))
+		if err != nil {
+			return nil, err
+		}
+
+		locale := strings.Split(fileName, ".")[0]
+
+		if IsDefaultLocale(fileName) {
+			defaultLocale = locale
+		}
+
+		translations[locale] = data
+	}
+
+	if len(translations) == 0 {
+		return nil, nil
+	} else {
+		return &core.Localization{
+			DefaultLocale: defaultLocale,
+			Translations:  translations,
+		}, nil
+	}
+}
+
+func GetFileNames(folderPath string) ([]string, error) {
+	files := []string{}
+	items, err := ioutil.ReadDir(folderPath)
+	if err != nil {
+		return files, err
+	}
+	for _, item := range items {
+		if !item.IsDir() {
+			files = append(files, item.Name())
+		}
+	}
+	return files, nil
+}
+
+func GetMapFromFile(filePath string) (map[string]interface{}, error) {
+	var result map[string]interface{}
+
+	jsonFile, err := os.Open(filePath)
+
+	if err != nil {
+		return result, err
+	}
+
+	defer jsonFile.Close()
```

Comment:
You had a comment in here that added some helpful context. I think it was something about needing to parse the file later on ... ?